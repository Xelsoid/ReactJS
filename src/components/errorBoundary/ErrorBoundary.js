import React from 'react';
import PropTypes from 'prop-types';

export default class ErrorBoundary extends React.Component {
  constructor() {
    super();
    this.state = {
      hasError: false,
    };
  }

  componentDidCatch(error, errorInfo) {
    console.log(`${error} : ${errorInfo}`); // eslint-disable-line no-console
    this.setState({hasError: true});
  }

  render() {
    const {children} = this.props;
    const {hasError} = this.state;
    if (hasError) {
      return (
        <div>Все поломалось! Но мы починим! Но это не точно!</div>
      );
    }
    return children;
  }
}
ErrorBoundary.defaultProps = {
  children: null
};

ErrorBoundary.propTypes = {
  children: PropTypes.instanceOf(Object),
};
