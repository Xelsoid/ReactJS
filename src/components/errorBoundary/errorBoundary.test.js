import React from 'react';
import ErrorBoundary from './ErrorBoundary';

it('renders correctly', () => {
  const component = shallow(
    <ErrorBoundary />
  );
  expect(component.state('hasError')).toEqual(false);
  expect(component).toMatchSnapshot();
});

it('renders correctly Error', () => {
  const component = shallow(
    <ErrorBoundary />
  );
  component.setState({hasError: true});
  expect(component.state('hasError')).toEqual(true);
  expect(component).toMatchSnapshot();
});
