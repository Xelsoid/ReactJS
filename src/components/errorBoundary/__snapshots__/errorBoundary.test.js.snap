// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `""`;

exports[`renders correctly Error 1`] = `
<div>
  Все поломалось! Но мы починим! Но это не точно!
</div>
`;
